core/shared/src/main/scala/cats/effect/IO.scala

```diff
@@ -1963,6 +1963,9 @@ object IO extends IOCompanionPlatform with IOLowPriorityImplicits {
   // later in the file
   private[this] val _never: IO[Nothing] = asyncForIO.never
 
+  implicit def secureRandom: SecureRandom[IO] =
```

Comment:
Otherwise it will create a new one each time :)

```suggestion
  lazy val secureRandom: SecureRandom[IO] =
```

---

Call-out for other reviewers: are we okay about unsafely creating a global secure random like this? I think having this would greatly enhance UX, but it creates a muddy precedent to use a pattern that has gone wrong sometimes e.g. FS2 `Network` ...

Reply:
Hmm, how to add the implicit to scope having it as lazy val?

```
[warn] /Users/diogocanut/Projects/cats-effect/core/shared/src/main/scala/cats/effect/IO.scala:1447:37: method fromSync in trait UUIDGenCompanionPlatformLowPriority is deprecated (since 3.6.0): Put an implicit `SecureRandom.javaSecuritySecureRandom` into scope to get a more efficient `UUIDGen`, or directly call `UUIDGen.fromSecureRandom`
[warn]   def randomUUID: IO[UUID] = UUIDGen[IO].randomUUID
```

Reply:
Oh sorry, I meant to write `implicit lazy val` :)

Reply:
@armanbilge What was the problem in FS2 `Network`?

Reply:
Well, it's a rather involved discussion :) so the problem is that FS2 promises an implicit `Network[IO]`, but this is technically encapsulating some global resources (namely, I/O threads). This was working well enough, until we started the polling system work. It created a significant challenge, since now we have to thread the `IOApp`s polling system into this globally promised `Network[IO]`.

If instead FS2 had been offering a `Resource[IO, Network[IO]]` that you had to construct explicitly, it would have been a lot easier to work with.

The fundamental question here is, when is it okay to "cheat" and unsafely allocate global shared state or resources, instead of forcing it to be explicitly allocated in `IO` or `Resource` by the user. It's some kind of UX trade-off.

See a similar place where we faced this in:
- https://github.com/typelevel/cats-effect/pull/3465

Reply:
Thanks, I think I understand. Btw, I'm not convinced a `Random[IO]` must be implicitly available: there isn't "one obvious preferable" `Random[IO]`. However, for `SecureRandom`, there could be: in fact, this PR tries to do that... so I think I'm okay with it in this specific case. (But it's possible i just don't appreciate the possible future pain this could cause :-)